src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/ModuleTokenResolver.cs

```diff
@@ -133,86 +157,91 @@ public void AddModuleTokenForType(TypeDesc type, ModuleToken token)
             }
         }
 
-        private class GenericTypeProvider : ISignatureTypeProvider<EntityHandle, ModuleTokenResolver>
+        private class TokenResolverProvider : ISignatureTypeProvider<bool, ModuleTokenResolver>
```

Comment:
We could use an opaque type here to indicate that no, the return values mean nothing with how we're using this class.

Reply:
That was my thinking too. Do we have something like a standard class Void?

Reply:
Frankly speaking, I'm not yet 100% sure this concept will fly longer-term, we may need to keep the elements in order to be able to construct the composite objects but right now I don't see any use for that so I enjoy keeping that as simple as possible.